icicle-source/src/Icicle/Source/Checker/Invariants.hs

```diff
@@ -48,16 +48,13 @@ invariantQ ctx (Query (c:cs) xfinal)
      -> errBanLatest
 
     GroupBy _ x
-     -> goX x >> goBanWindowAndGroupFold
+     -> goX x >> go
 
     Distinct _ x
-     -> goX x >> goBanWindowAndGroupFold
+     -> goX x >> go
 
     GroupFold _ _ _ x
-     | allowWindows inv
-     -> goX x >> goBanWindowAndGroupFold
-     | otherwise
-     -> errBanGroupFold
```

Comment:
@amosr I think it should be stricter than this, but what we had before was too strict?